test/test_utils/security.py

```diff
@@ -1,13 +1,133 @@
+        *args: Any,
+        **kwargs: Any,
+    ):
+        self.file_path = filePath
+        self.name = name
+        self.package_manager = packageManager
+        self.version = version
+        self.release = release
+
+
+@dataclass
+class AllowListFormatVulnerability:
+    """
+    AllowlistFormatVulnerability represents how the data looks for a single vulnerability in the allowlist format.
+    The data from the ECR Ehanced Results are deserialized into AllowListFormatVulnerability dataclass. In 
+    other words, vulnerabilities from the ecr format are are directly deserialized into vulnerabilities in Allowlist
+    format using AllowlistFormatVulnerability dataclass.
+    """
+
+    description: str
+    vulnerability_id: str
+    name: str
+    package_name: str
+    package_details: VulnerablePackageDetails
+    remediation: dict
+    cvss_v3_score: float
+    cvss_v30_score: float
+    cvss_v2_score: float
+    cvss_v3_severity: str
+    source_url: str
+    source: str
+    severity: str
+    status: str
+    title: str
+
+    def __init__(
+        self,
+        description: str,
+        packageVulnerabilityDetails: dict,
+        remediation: dict,
+        severity: str,
+        status: str,
+        title: str,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        self.description = description
+        self.vulnerability_id = packageVulnerabilityDetails["vulnerabilityId"]
+        self.name = packageVulnerabilityDetails["vulnerabilityId"]
+        self.package_name = None
+        self.package_details = None
+        self.remediation = remediation
+        self.source_url = packageVulnerabilityDetails["sourceUrl"]
+        self.source = packageVulnerabilityDetails["source"]
+        self.severity = severity
+        self.status = status
+        self.title = title
+        self.cvss_v3_score = self.get_cvss_score(packageVulnerabilityDetails, score_version="3.1")
+        self.cvss_v30_score = self.get_cvss_score(packageVulnerabilityDetails, score_version="3.0")
+        self.cvss_v2_score = self.get_cvss_score(packageVulnerabilityDetails, score_version="2.0")
+        self.cvss_v3_severity = self.get_cvss_v3_severity(self.cvss_v3_score)
+
+    def get_cvss_score(self, packageVulnerabilityDetails: dict, score_version: str = "3.1"):
+        for cvss_score in packageVulnerabilityDetails["cvss"]:
+            if cvss_score["version"] == score_version:
+                return float(cvss_score["baseScore"])
+        return 0.0
+
+    ## Taken from https://nvd.nist.gov/vuln-metrics/cvss and section 5 of first.org/cvss/specification-document
+    def get_cvss_v3_severity(self, cvss_v3_score: float):
+        if cvss_v3_score >= 9.0:
+            return "CRITICAL"
+        elif cvss_v3_score >= 7.0:
+            return "HIGH"
+        elif cvss_v3_score >= 4.0:
+            return "MEDIUM"
+        elif cvss_v3_score >= 0.1:
+            return "LOW"
+        return "UNDEFINED"  # Used to represent None Severity as well
```

Comment:
Please use enum or class-type variables, such as `CVESeverity`, rather than using string literals, to prevent bugs caused by spelling mistakes in the future.

https://github.com/aws/deep-learning-containers/blob/78ab7cbfe0305246c12f663018a961ec77e3c60b/test/test_utils/security.py#L21-L27

Reply:
I want the values to be actual strings and not their integer values defined by CVESeverity.

Reply:
This is okay for now, but you can use `CVESeverity.CRITICAL.name`, for example, to get the string rather than the number. The number, instead, can be obtained using `CVESeverity.CRITICAL.value`.